File: leetcode/0136. Single Number/Solution.swift
// https://leetcode.com/problems/single-number/

class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
        return nums.reduce(0, ^)
    }
}

let s = Solution()
assert(s.singleNumber([2,2,1]) == 1)
assert(s.singleNumber([4,1,2,1,2]) == 4)

File: leetcode/0136. Single Number/Solution.py
# https://leetcode.com/problems/single-number/

from typing import List
from functools import reduce

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return reduce(lambda x, y: x ^ y, nums)

s = Solution()
assert s.singleNumber([2,2,1]) == 1
assert s.singleNumber([4,1,2,1,2]) == 4
